ext/vulnsrc/rhel/rhel.go

```diff
@@ -186,16 +187,22 @@ func parseRHSA(ovalReader io.Reader) (vulnerabilities []database.VulnerabilityWi
 		if len(pkgs) > 0 {
 			vulnerability := database.VulnerabilityWithAffected{
 				Vulnerability: database.Vulnerability{
-					Name:        name(definition),
-					Link:        link(definition),
 					Severity:    severity(definition),
 					Description: description(definition),
 				},
 			}
 			for _, p := range pkgs {
 				vulnerability.Affected = append(vulnerability.Affected, p)
 			}
-			vulnerabilities = append(vulnerabilities, vulnerability)
+
+			// One vulnerability by CVE
+			for _, reference := range definition.References {
```

Comment:
Let's put this logic in the name function.

Also, let's try to do this and if there isn't a Source reference, we can failover to the RHSA value.

The same logic can be applied to links in the link function.

Reply:
I think that the name function was convenient when there was only one vulnerability to deal with.

This for loop creates one vulnerability for each CVE referenced by the RHSA (ie: (len(references) - 1) vulnerabilities).

Maybe I could iterate on definition.References on line 188 and call name and link with the reference instead of the definition?

Something like:

	for _, definition := range ov.Definitions {
		pkgs := toFeatures(definition.Criteria)
		if len(pkgs) > 0 {
			var affected []database.AffectedFeature
			for _, p := range pkgs {
				affected = append(affected, p)
			}

			for _, reference := range definition.References {
				if reference.Source == "CVE" {
					vulnerability := database.VulnerabilityWithAffected{
						Vulnerability: database.Vulnerability{
							Severity:    severity(definition),
							Description: description(definition),
							Name:        name(reference),
							Link:           link(reference),
						},
						Affected: affected,
					}
					vulnerabilities = append(vulnerabilities, vulnerability)
				}
			}
		}
	}

And then failover to a unique vulnerability with RHSA ID if there are no CVE at all.

Reply:
That sounds reasonable.